modules/dnn/src/dnn.cpp

```diff
@@ -2458,7 +2458,7 @@ struct Net::Impl : public detail::NetImplBase
                                     if( nextData )
                                         nextActivLayer = nextData->layerInstance.dynamicCast<ActivationLayer>();
 
-                                    if( !nextActivLayer.empty() && pinsToKeep.count(lpNext) == 0 &&
+                                    if( !nextActivLayer.empty() &&
```

Comment:
This change need not be propagated to master. The eltwise fusion logic is divergent. #17939 applies the same fix on master.